torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -401,105 +447,119 @@ void ProcessGroupAgent::enqueueSend(SendWork work) {
           handleSend(work);
         } catch (std::exception& e) {
           if (work.message_.isRequest()) {
-            std::ostringstream ss;
-            ss << "Encountered exception in ProcessGroupAgent::enqueueSend: "
-               << e.what();
+            auto err = c10::str(
+                "Encountered exception in ProcessGroupAgent::enqueueSend: ",
+                e.what());
             auto exceptionMsg =
-                rpc::createExceptionResponse(work.message_, ss.str());
+                rpc::createExceptionResponse(err, work.message_.id());
             markFutureWithError(exceptionMsg);
           }
         }
       },
       std::move(work)));
 }
 
-void ProcessGroupAgent::enqueueRecv(RecvWork work) {
-  threadPool_.run(std::bind(
-      [&](RecvWork& work) {
-        torch::Tensor& payload = work.payload_;
-        auto data = wireDeserialize(payload.storage().data(), payload.numel());
-        Message message(
-            std::move(data.first),
-            std::move(data.second),
-            work.type_,
-            work.id_);
-        if (message.isRequest()) {
-          auto futureResponse = cb_->operator()(message);
-          if (futureResponse->completed()) {
-            if (!futureResponse->hasError()) {
-              send(work.from_, std::move(*futureResponse).moveValue());
+void ProcessGroupAgent::handleRecv(RecvWork& work) {
```

Comment:
for reviewers: there are no real changes in this logic, the code is just moved to a helper function.